src/cephadm/cephadm

```diff
@@ -4788,21 +4788,21 @@ def command_check_host():
             find_program(command)
             logger.info('%s is present' % command)
         except ValueError:
-            errors.append('ERROR: %s binary does not appear to be installed' % command)
+            errors.append('%s binary does not appear to be installed' % command)
 
     # check for configured+running chronyd or ntp
     if not check_time_sync():
-        errors.append('ERROR: No time synchronization is active')
+        errors.append('No time synchronization is active')
 
     if 'expect_hostname' in args and args.expect_hostname:
         if get_hostname().lower() != args.expect_hostname.lower():
-            errors.append('ERROR: hostname "%s" does not match expected hostname "%s"' % (
+            errors.append('hostname "%s" does not match expected hostname "%s"' % (
                 get_hostname(), args.expect_hostname))
         logger.info('Hostname "%s" matches what is expected.',
                     args.expect_hostname)
 
     if errors:
-        raise Error('\n'.join(errors))
+        raise Error('\nERROR: '.join(errors))
```

Comment:
not going to work. this would look like
```
error1
ERROR: error2
ERROR: error3
ERROR: error4
```

What about 

```python
print({'errors': errors}
```

And change

https://github.com/ceph/ceph/blob/beb00bd2a8ecc486fc6ddbe00cf9e4cb4bc20509/src/pybind/mgr/cephadm/module.py#L1192

to 

```python
errors = json.loads(out)['errors']
```

Reply:
ehh the handling of the `Error` class adds it:
https://github.com/ceph/ceph/blob/7f75a745b76820e0aec4b3e8ed22e864f0737308/src/cephadm/cephadm#L7286



Adding structured output (e.g. `--format json`) would actually be a better solution, but that's an issue with many of the cephadm commands:
https://github.com/ceph/ceph/pull/37712#issuecomment-722631095

Reply:
hm. We already commands that properly return json, like so:

https://github.com/ceph/ceph/blob/71711b56db1d595fb02171e0fa9a53c232546952/src/cephadm/cephadm#L4297-L4300

I think #37712 deals with the problem for `bootstrap` that the log output is **right now written to stderr** in order to avoid breaking the json output.

Reply:
yeah, but it feels odd to have normal log messages on stderr with only json formatted errors on stdout ...

Reply:
json formatted errors on stdout ...

in this case, the error messages are the payload, i.e. the result of the operation, *not* error message while performing the operation. This actually feels right for me :smile:

Reply:
I see. It's not that simple as we need to also handle other calls to `check_host()`